### PR TITLE
Use Zuora Tap

### PR DESCRIPTION
**Pull request by mhindery**
## Description

I want to use the [Zuora tap](https://github.com/singer-io/tap-zuora) with PPW, and hopefully this can be added to the upstream. I got the basics working with little effort.

As far as PPW goes, this is the changes that are required to use it, which are as described in the contribution docs. There was a step missing from that doc in general though, so I've added that one here. It does not include tests or documentation or anything else yet for this tap, should that be required.

The Zuora tap checks of the requirements described in the [Adding a new tap](https://transferwise.github.io/pipelinewise/project/contribution.html) document. However when running it, it turns out that it didn't include an entry with empty breadcrumbs in its schema. I compared the tap's catalog file with the one postgres has, and it looked lik that one entry just needed to be added for each stream. So I forked the tap and added that, which actually is just a single line change: [PR of my fork with added line](https://github.com/OTA-Insight/tap-zuora/pull/1/files). That fork is the one which is installed in this PR as well instead of the official package. The tap also does not include all columns by default, and leaves the choice on what to select up to the user. PPW does not support this at this moment, so it has been changed that all available columns are exported in the tap. I suppose if this gets accepted, that fork with these changes would become a PPW-customized fork just like the other taps have.

A config yaml for this tap which I'm using looks like this:

```yml
---
id: "zuora"
name: "Zuora"
type: "tap-zuora"
owner: "mathieu@otainsight.com"

db_conn:
  username: "{USERNAME}"
  password: "{PASSWORD}"
  api_type: "AQUA"  # or "REST"
  start_date: "2020-04-01"
  sandbox: "false"
  european: "true"
  partner_id: "{PARTNER_ID}"

target: "postgres_dwh_dev"
batch_size_rows: 4000
default_target_schema: "zuora"

schemas:
  - source_schema: "zuora"
    target_schema: "zuora"
    tables:
    - table_name: "Account"
      replication_method: "INCREMENTAL"
      replication_key: "updatedDate"
    - table_name: "BillingRun"
      replication_method: "INCREMENTAL"
      replication_key: "updatedDate"
```

This succesfully runs and loads data.

## Checklist

- [ ] Description above provides context of the change
- [ ] Unit tests coverage for changes (not needed for documentation changes)
- [ ] Bumping version in `setup.py` is an individual PR and not mixed with feature or bugfix PRs
- [ ] Commit message/PR title starts with `[AP-NNNN]` if applicable. AP-NNNN = JIRA ID
- [ ] Branch name starts with `AP-NNN` if applicable. AP-NNN = JIRA ID
- [ ] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [ ] Relevant documentation is updated including usage instructions

### CODE REVIEW

**Comment by koszti** [comment]
Thanks for making this. I will create `pipelinewise-tap-zuora` using your fork as the baseline and will publish to PyPI. In the meantime please add the documenation 

* **Register Connector**. New `tap-zuora` dir with pypi requirements [here](https://github.com/transferwise/pipelinewise/tree/master/singer-connectors)
* **Documentation**. New `zuora.rst` doc [here](https://github.com/transferwise/pipelinewise/tree/master/docs/connectors/taps) and [here](https://github.com/transferwise/pipelinewise/blob/master/docs/connectors/taps.rst): Add YAML example and every required information that is required to set up up zuora connection. Also an image is nice to have if the logo is not copyright licensed and can be shown in external sites like this one. You'll find examples in other rst files
* **Sample YAML files**. New`tap_zuora.yml.sample` file [here](https://github.com/transferwise/pipelinewise/tree/master/pipelinewise/cli/samples): This will be generated when creating a new project
* **Install Script**. [install.sh](https://github.com/transferwise/pipelinewise/blob/e9147c06f490da1404db6d56f696370dba147e26/install.sh#L178): Please add tap-zuora as an extra connector for now
* **Github Readme**. [README.md](https://github.com/transferwise/pipelinewise/blob/master/README.md): Add zuora to the connectors list
* Any test cases are very welcome

**Comment by mhindery** [comment]
@koszti I have added some of the requested items, all documentation and sample yamls have been added.

- For the requirements file of the custom tap, I've used `pipelinewise-tap-zuora==1.0.0`. Don't know what the version will be that you make, let me know if I need to change it.
- What kind of tests are expected or required for taps? Any particular example I can look at and modify?

**Comment by koszti** [comment]
Yes, `pipelinewise-tap-zuora==1.0.0` in requirements.txt is, thank you.
I sent a PR to your fork with some standard PPW items that's required before moving it into the final place. Please review and approve/merge when you're happy with it: https://github.com/OTA-Insight/tap-zuora/pull/3

**Comment by mhindery** [comment]
I've merged the MR.

File: README.md [hunk 1]
```diff
@@ -45,7 +49,8 @@ consumes data from taps and do something with it, like load it into a file, API
...
-| Tap       | **[Salesforce](https://github.com/singer-io/tap-salesforce)** | [![PyPI version](https://badge.fury.io/py/tap-salesforce.svg)](https://badge.fury.io/py/tap-zendesk) | Extracts data from Salesforce database using BULK and REST extraction API with Key-Based incremental replications |
+| Tap       | **[Salesforce](https://github.com/singer-io/tap-salesforce)** | [![PyPI version](https://badge.fury.io/py/tap-salesforce.svg)](https://badge.fury.io/py/tap-salesforce) | Extracts data from Salesforce database using BULK and REST extraction API with Key-Based incremental replications |
+| Tap       | **[Zuora](https://github.com/singer-io/tap-zuora)** | [![PyPI version](https://badge.fury.io/py/tap-zuora.svg)](https://badge.fury.io/py/tap-zuora) | Extracts data from Zuora database using AQAA and REST extraction API with Key-Based incremental replications |
```
Comment (koszti) [inline README.md:53]: Can you please change the link to:
* https://github.com/transferwise/pipelinewise-tap-zuora
* https://badge.fury.io/py/pipelinewise-tap-zuora.svg
* https://badge.fury.io/py/pipelinewise-tap-zuora

**Comment by koszti** [comment]
created the fork repo at https://github.com/transferwise/pipelinewise-tap-zuora and published to PyPI to https://pypi.org/project/pipelinewise-tap-zuora/ .

Please make the small change request above and we can merge.

File: README.md (hunk 1 above)
Comment (mhindery) [reply README.md]: Has been updated.

**Pull request merged**